src/main/java/uk/ac/cam/cl/dtg/segue/etl/ContentIndexer.java

```diff
@@ -629,42 +636,63 @@ private synchronized void buildElasticSearchIndex(final String sha,
             }
         }
 
+        long startTime, endTime;
 
         try {
             es.indexObject(sha, CONTENT_INDEX_TYPE.METADATA.toString(),
                     objectMapper.writeValueAsString(ImmutableMap.of("version", sha, "created", new Date().toString())), "general");
             es.indexObject(sha, CONTENT_INDEX_TYPE.METADATA.toString(),
                     objectMapper.writeValueAsString(ImmutableMap.of("tags", tagsList)), "tags");
 
-            // TODO: Should probably bulk index these
-            for (String k : allUnits.keySet()) {
-                es.indexObject(sha, CONTENT_INDEX_TYPE.UNIT.toString(),
-                        objectMapper.writeValueAsString(ImmutableMap.of("cleanKey", k, "unit", allUnits.get(k))));
-            }
-            for (String k : publishedUnits.keySet()) {
-                es.indexObject(sha, CONTENT_INDEX_TYPE.PUBLISHED_UNIT.toString(),
-                        objectMapper.writeValueAsString(ImmutableMap.of("cleanKey", k, "unit", publishedUnits.get(k))));
-            }
+            startTime = System.nanoTime();
+            es.bulkIndex(sha, CONTENT_INDEX_TYPE.UNIT.toString(), allUnits.entrySet().stream().map(entry -> {
+                try {
+                    return immutableEntry((String) null, objectMapper.writeValueAsString(ImmutableMap.of("cleanKey", entry.getKey(), "unit", entry.getValue())));
+                } catch (JsonProcessingException jsonProcessingException) {
+                    log.error("Unable to serialise unit entry for unit: " + entry.getValue());
+                    return null;
+                }
+            }).filter(Objects::nonNull).collect(Collectors.toList()));
```

Comment:
The filter is nice given that Java doesn't seem to have a particularly good way of combining checked exceptions and streams.